ts/editor/EditorField.svelte

```diff
@@ -0,0 +1,104 @@
+<!--
+Copyright: Ankitects Pty Ltd and contributors
+License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+-->
+<script context="module" lang="ts">
+    import type { EditingAreaAPI } from "./EditingArea.svelte";
+    import contextProperty from "../sveltelib/context-property";
+
+    export interface FieldData {
+        name: string;
+        fontFamily: string;
+        fontSize: number;
+        direction: "ltr" | "rtl";
+    }
+
+    export interface EditorFieldAPI {
+        element: HTMLElement;
+        index: number;
+        direction: "ltr" | "rtl";
+        editingArea?: EditingAreaAPI;
+    }
+
+    const key = Symbol("editorField");
+    const [set, getEditorField, hasEditorField] = contextProperty<EditorFieldAPI>(key);
+
+    export { getEditorField, hasEditorField };
+</script>
+
+<script lang="ts">
+    import EditingArea from "./EditingArea.svelte";
+    import LabelContainer from "./LabelContainer.svelte";
+    import LabelName from "./LabelName.svelte";
+    import FieldState from "./FieldState.svelte";
+
+    import { setContext } from "svelte";
+    import { writable } from "svelte/store";
+    import type { Writable } from "svelte/store";
+    import { directionKey } from "../lib/context-keys";
+
+    export let content: Writable<string>;
+    export let field: FieldData;
+    export let autofocus = false;
+
+    const directionStore = writable();
+    setContext(directionKey, directionStore);
+
+    $: $directionStore = field.direction;
+
+    let editorField: HTMLElement;
+
+    export const api = set(
+        Object.create(
+            {},
+            {
+                element: {
+                    get: () => editorField,
+                },
+                direction: {
+                    get: () => $directionStore,
+                },
+            }
+        ) as EditorFieldAPI
```

Comment:
Just out of curiosity, why is Object.create required here?

Reply:
So this:
```js
        element: {
            get: () => editorField,
        },
```

is necessary, because at the time of component instantiation, we don't have access to the DOM element `editorField` yet. Using `Object.create` (or `Object.defineProperties`) basically allows lazy loading these. This means you can use `editorField.element`, instead of `editorField.getElement()`. Same for `.direction` instead of `.direction()`. 

Both of these are up for refactoring very soon though. For any field preferences, I probably want to publish them as one object on editor field, and for `element, I already mentioned above that I probably should use promises for this.